### PR TITLE
Pu/issue 441

### PR DESCRIPTION
**Pull request by izulin**
### Context

Add two helper functions:
getCellDependents
and
getCellDependencies

when queried with address or simple range of addresses, returns list of addresses/ranges that:
- depend on arg for computation
- computation of arg depends on them

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #441 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: src/HyperFormula.ts [hunk 1]
```diff
@@ -2191,6 +2191,60 @@ export class HyperFormula implements TypedEmitter {
...
+  /**
+   * Returns all addresses and ranges whose computation depends on input address or range provided.
+   *
+   * @param {SimpleCellAddress | AbsoluteCellRange} address - object representation of an absolute address or range of addresses
+   *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray( [ ['1', '=A1', '=A1+B1'] ] );
+   * hfInstance.getCellDependents({ sheet: 0, col: 0, row: 0});
+   * // should return [{ sheet: 0, col: 1, row: 0}, { sheet: 0, col: 2, row: 0}]
```
Comment (wojciechczerniak) [inline src/HyperFormula.ts:2203]: code block is not closed
```suggestion
   * ```js
   * const hfInstance = HyperFormula.buildFromArray( [ ['1', '=A1', '=A1+B1'] ] );
   *
   * hfInstance.getCellDependents({ sheet: 0, col: 0, row: 0});
   * // should return [{ sheet: 0, col: 1, row: 0}, { sheet: 0, col: 2, row: 0}]
   * ```
```

File: src/HyperFormula.ts [hunk 2]
```diff
@@ -2191,6 +2191,60 @@ export class HyperFormula implements TypedEmitter {
...
+  /**
+   * Returns all addresses and ranges necessary for computation of a given address or range.
+   *
+   * @param {SimpleCellAddress | AbsoluteCellRange} address - object representation of an absolute address or range of addresses
+   *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray( [ ['1', '=A1', '=A1+B1'] ] );
+   * hfInstance.getCellDependents({ sheet: 0, col: 2, row: 0});
+   * // should return [{ sheet: 0, col: 0, row: 0}, { sheet: 0, col: 1, row: 0}]
```
Comment (wojciechczerniak) [inline src/HyperFormula.ts:2230]: code block is not closed
```suggestion
   * ```js
   * const hfInstance = HyperFormula.buildFromArray( [ ['1', '=A1', '=A1+B1'] ] );
   *
   * hfInstance.getCellDependents({ sheet: 0, col: 2, row: 0});
   * // should return [{ sheet: 0, col: 0, row: 0}, { sheet: 0, col: 1, row: 0}]
   * ```
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Added helper functions for keeping track of cell/range dependencies.
```
Comment (wojciechczerniak) [inline CHANGELOG.md:8]:
```suggestion

### Added
- Added helper functions for keeping track of cell/range dependencies. (#441)
```

File: src/HyperFormula.ts [hunk 4]
```diff
@@ -2191,6 +2191,64 @@ export class HyperFormula implements TypedEmitter {
...
+  /**
+   * Returns all addresses and ranges necessary for computation of a given address or range.
+   *
+   * @param {SimpleCellAddress | AbsoluteCellRange} address - object representation of an absolute address or range of addresses
+   *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray( [ ['1', '=A1', '=A1+B1'] ] );
+   *
+   * hfInstance.getCellDependents({ sheet: 0, col: 2, row: 0});
```
Comment (wojciechczerniak) [inline src/HyperFormula.ts:2232]: wrong method is used in the example
```suggestion
   * hfInstance.getCellDependencies({ sheet: 0, col: 2, row: 0});
```

**Comment by scarletfog** [comment]
@wojciechczerniak when I peek into the changelog file it says:
"Added **helper** functions for keeping track of cell/range dependencies"

So maybe just... "Helpers"?  "Tracers" sounds a bit weird to me. "Graph helpers"? 🤔

**Comment by wojciechczerniak** [comment]
@scarletfog @izulin "Helpers" sounds good

File: src/HyperFormula.ts [hunk 5]
```diff
@@ -2191,6 +2191,68 @@ export class HyperFormula implements TypedEmitter {
...
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray( [ ['1', '=A1', '=A1+B1'] ] );
+   *
+   * hfInstance.getCellDependencies({ sheet: 0, col: 2, row: 0});
+   * // should return [{ sheet: 0, col: 0, row: 0}, { sheet: 0, col: 1, row: 0}]
+   * ```
+   *
+   * @category Helpers
+   */
+  public getCellDependencies(address: SimpleCellAddress | AbsoluteCellRange): (AbsoluteCellRange | SimpleCellAddress)[] {
```
Comment (wojciechczerniak) [inline src/HyperFormula.ts:2240]: Let's name it as other spreadsheets do. They have `Trace Dependents` and `Trace Precedents`. Will be easier to distinguish both methods.

```suggestion
  public getCellPrecedents(address: SimpleCellAddress | AbsoluteCellRange): (AbsoluteCellRange | SimpleCellAddress)[] {
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- Added helper functions for keeping track of cell/range dependencies. (#441)
```
Comment (wojciechczerniak) [inline CHANGELOG.md:10]: We should mention the new methods by their names

File: src/HyperFormula.ts (hunk 5 above)
Comment (doczoidberg) [reply src/HyperFormula.ts]: is there any time estimation on this feature?

Comment (doczoidberg) [reply src/HyperFormula.ts]: we also need the dependents

Comment (izulin) [reply src/HyperFormula.ts]: I left a reply in issue #487 :)

**Pull request merged**